test/NinjaTools.FluentMockServer.API.Tests/Downstream/RequestMapperSpec.cs

```diff
@@ -0,0 +1,355 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Threading.Tasks;
+using FluentAssertions;
+using JetBrains.Annotations;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Primitives;
+using NinjaTools.FluentMockServer.API.Downstream;
+using NinjaTools.FluentMockServer.API.Extensions;
+using NinjaTools.FluentMockServer.API.Extensions.Responses;
+using NinjaTools.FluentMockServer.Tests.TestHelpers;
+using NinjaTools.FluentMockServer.Tests.TestHelpers.Data.TheoryData;
+using Shouldly;
+using TestStack.BDDfy;
+using TestStack.BDDfy.Xunit;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace NinjaTools.FluentMockServer.API.Tests.Downstream
+{
+    // [ExampleData(ExampleData = typeof(EmailExampleTable))]
+    [Story( TitlePrefix = "Internal",Title = "Modify scoped HttpRequest before they get routed by MVC.",
+        AsA = "API user",
+        IWant = "I want to configure httpRequest re-routes",
+        SoThat = "I can setup up actions the mockserver should take.")]
+    public class RequestMapperSpec : XUnitTestBase<RequestMapperSpec>
+    {
+        private HttpRequest _inputRequest;
+        private Response<HttpRequestMessage> _result;
+        private readonly RequestMapper _requestMapper;
+        private List<KeyValuePair<string, StringValues>> _inputHeaders;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused private field '_inputHeaders'.](https://app.codacy.com/app/git_36/NinjaTools.FluentMockServer/pullRequest?prid=4722826)